modules/imgproc/src/intersection.cpp

```diff
@@ -51,10 +51,20 @@ int rotatedRectangleIntersection( const RotatedRect& rect1, const RotatedRect& r
 {
     CV_INSTRUMENT_REGION();
```

Comment:
There are several return statements from this code.
Some of them are **NOT** handled properly (e.g., line 97/107).

Robust way is to to create wrapper which would call the original code:
- make existed implementation as internal:
  ```
  static
  int rotatedRectangleIntersection_(
          const RotatedRect& rect1,
          const RotatedRect& rect2,
          CV_OUT std::vector <Point2f>& intersection)
  {
      ... almost the same existed code ...
  }
  ```
- add wrapper below which shifts input and adjusts results:
  ```
  int rotatedRectangleIntersection(const RotatedRect& rect1, const RotatedRect& rect2, OutputArray intersectingRegion)
  {
      CV_TRACE_FUNCTION();
      ... shift input ...
      std::vector <Point2f> intersection; intersection.reserve(24);
      int result = rotatedRectangleIntersection_(shiftedRect1, shiftedRect2, intersection);
      ... adjust 'intersection' points ...
      Mat(intersection).copyTo(intersectingRegion);  // ??? handle .empty() case
      return result;
  }
  ```